docs/content/topics/operators/gateways.md

```diff
@@ -11,340 +11,518 @@ aliases:
-      project: ${project_id}
-      build: ${uuid}
-      component: build
-  type: "brigade.sh/build"
-  data:
-    revision:
-      commit: $("${base64[@]}" <<<"${commit_id}")
-      ref: $("${base64[@]}" <<<"${commit_ref}")
-    event_provider: $("${base64[@]}" <<<"${event_provider}")
-    event_type: $("${base64[@]}" <<<"${event_type}")
-    project_id: $("${base64[@]}" <<<"${project_id}")
-    build_id: $("${base64[@]}" <<<"${uuid}")
-    payload: $("${base64[@]}" <<<"${payload}")
-    script: $("${base64[@]}" <<<"${script}")
-EOF
-  sleep 60
-done
-```
+Since the Brigade API server is the point of contact for gateway
+authentication/authorization and event submission, gateway developers will need
+to pick an [SDK] to use. For this example, we'll be using the [Go SDK]. As of
+this writing, a [Javascript/Typescript SDK] and a [Rust SDK] also exist.
 
-While the main point of the script above is just to show how to create a basic
-event, it should also demonstrate how flexible the system is. A script can take
-input from just about anything and use it to trigger a new event.
+[SDK]: https://github.com/brigadecore/brigade/tree/v2#sdks
+[Go SDK]: https://github.com/brigadecore/brigade/tree/v2/sdk
+[Javascript/Typescript SDK]: https://github.com/brigadecore/brigade-sdk-for-js
+[Rust SDK]: https://github.com/brigadecore/brigade-sdk-for-rust
 
-## Creating A Cron Job Gateway
+## Example Gateway
 
-Beginning with the code above, we can build a gateway that runs as a scheduled
-job in Kubernetes. In this example, we use a Kubernetes
-[CronJob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) object
-to create the secret.
+The following example assumes a running Brigade instance deployed with the
+default root user enabled. If you'd like to follow along and haven't yet
+deployed Brigade, check out the [QuickStart].
 
-First we can begin with a simplified version of the script above. This one does not
-run in a loop. It just runs once to completion.
+[QuickStart]: /intro/quickstart
 
-Here is `cron-event.sh`:
+### Authentication
 
-```bash
-#!/usr/bin/env bash
-set -euo pipefail
+As a first course of action for our example gateway written in Go, we'll need
+to authenticate with Brigade. Here is what the Go code looks like, with in-line
+comments describing each section:
 
-# The Kubernetes namespace in which Brigade is running.
-namespace=${NAMESPACE:-default}
+```go
+package main
 
-event_provider="simple-event"
-event_type="my_event"
-project_id="brigade-830c16d4aaf6f5490937ad719afd8490a5bcbef064d397411043ac"
-commit_ref="master"
-commit_id="589e15029e1e44dee48de4800daf1f78e64287c0"
-uuid="$(uuidgen | tr '[:upper:]' '[:lower:]')"
-name="simple-event-$uuid"
+import (
+  "context"
+  "log"
 
-payload=$(date)
-script=$(cat <<EOF
-const { events } = require("brigadier");
-events.on("my_event", (e) => {
-  console.log("The system time is " + e.payload);
-});
-EOF
+  brigadeOS "github.com/brigadecore/brigade-foundations/os"
```

Comment:
imho, I wouldn't show use of brigade-foundations.

If all we're doing is getting some env vars, I'd worry this is more likely to distract from the important bits than it is to save anyone much work.

Use `os` from stdlib?

Reply:
Good call; removed.